src/outline.rs

```diff
@@ -5,26 +5,34 @@
 //! although a stable wrapper may be added in `Database`.
 
 use crate::parser::as_str;
+use crate::parser::Comparer;
 use crate::parser::HeadingDef;
 use crate::parser::HeadingLevel;
 use crate::parser::SegmentId;
 use crate::parser::SegmentRef;
 use crate::parser::StatementAddress;
+use crate::parser::StatementIndex;
 use crate::parser::Token;
 use crate::segment_set::SegmentSet;
+use crate::tree::NodeId;
+use crate::tree::Tree;
+use crate::Database;
+use std::cmp::Ordering;
+use std::fmt::Display;
+use std::fmt::Formatter;
 use std::sync::Arc;
 
 #[derive(Debug, Default, Clone)]
```

Comment:
Does it actually make sense for this type to implement `Default`? It looks like `new` actually needs some information.

Reply:
Indeed, it does not make much sense. However the compiler complains that this is needed so that Tree<OutlineNode> can derive default, even if that default is just an empty `Vec`: strange.

This is the corresponding code snippet:
```
struct A {
    a: u8,
}

struct B<T> {
    b: T,
}

#[derive(Default)] 
struct C<T> {
    c: Vec<B<T>>,
}

#[derive(Default)] 
struct D {
    d: C<A>,
}
```

Reply:
You have to manually implement `Default` for `C` in that example (and for `Tree` in the present case). This is a known limitation, it is difficult to determine what the actual bounds on the impl are (especially in a macro, which doesn't have type information).

Reply:
Done in the latest commit!